packages/maskbook/src/plugins/RedPacket/SNSAdaptor/confirmRedPacketForm.tsx

```diff
@@ -92,14 +91,11 @@ export function ConfirmRedPacketForm(props: ConfirmRedPacketFormProps) {
                     </Grid>
                     <Grid item xs={6}>
                         <Typography variant="body1" color="textPrimary" align="right">
-                            {`${
-                                settings?.isRandom
-                                    ? formatBalance(new BigNumber(settings?.total ?? 0), settings?.token?.decimals ?? 0)
-                                    : formatBalance(
-                                          new BigNumber(settings?.total ?? 0).div(settings?.shares ?? 1),
-                                          settings?.token?.decimals ?? 18,
-                                      )
-                            } ${settings?.token?.symbol}`}
+                            <FormattedBalance
+                                value={settings?.total}
+                                decimals={settings?.token?.decimals}
+                                symbol={settings?.token?.symbol}
+                            />
```

Comment:
<img width="986" alt="CleanShot 2021-07-09 at 15 08 39@2x" src="https://user-images.githubusercontent.com/3842474/125038077-8cc88f80-e0c7-11eb-96a6-fe342340c989.png">
?????

Reply:
@lelenei you unhandling decimalz